lib/puppet/indirector/hiera.rb

```diff
@@ -18,6 +18,7 @@ def initialize(*args)
   def find(request)
     not_found = Object.new
     options = request.options
+    Puppet.debug "Performing a hiera indirector lookup of #{request.key} with options #{options.inspect}"
```

Comment:
Always write calls to Puppet.debug using a lambda - i.e.:
````
Puppet.debug {"Performing a hiera indirector lookup of #{request.key} with options #{options.inspect}"}
````
This is 60% faster when not running in debug mode even for a simple string, and even faster when (as in this case) there is string interpolation required to construct the debug message).

This is important here, since looking up class parameters is performance critical.